### PR TITLE
[FIX] Cryopod : Fixed the cause of runtime that breaks the MC ?

### PR DESCRIPTION
**Pull request by TorinoFermic**
## About The Pull Request

Before this PR, a robot in a robotic cryo will cause a runtime because the despawn_occupant() will call the parent which lands in the proc with human-only code causing a runtime unless the line `occupant_organs = H.organs | H.internal_organs` is behind the type check

The PR fixes the spammy runtime, not sure if it also breaks the MC beforehand.

Last thing, should this fix to be changed to a refactor of cryo code so we have /obj/machinery/cryopod/organic and original /obj/machinery/cryopod/robot, the /obj/machinery/cryopod will just simply call the generic removal code ?

### CODE REVIEW

**Comment by TorinoFermic** [comment]
On hold while I refactor this code into /mob/ procs

**Comment by TorinoFermic** [comment]
Done refactoring the code for now.

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -452,6 +452,48 @@ default behaviour is:
...
+			if(O.owner && O.owner.current)
+				to_chat(O.owner.current, SPAN_WARNING("You get the feeling your target is no longer within your reach..."))
+			qdel(O)
+
+	//Same for contract-based objectives.
+	for(var/datum/antag_contract/contract in GLOB.all_antag_contracts)
+		contract.on_mob_despawned(src.mind)
+
+	//Handle job slot/tater cleanup.
+	var/job = src.mind.assigned_role
```
Comment (MadmanMartian) [inline code/modules/mob/living/living.dm:471]: Might want to check if there's a mind first

Comment (TorinoFermic) [reply code/modules/mob/living/living.dm:471]: With if(src.mind) check ?

Comment (MadmanMartian) [reply code/modules/mob/living/living.dm:471]: yes

Comment (TorinoFermic) [reply code/modules/mob/living/living.dm]: Done

File: code/game/machinery/cryopod.dm [hunk 2]
```diff
@@ -280,9 +286,12 @@
 // Also make sure there is a valid control computer
 /obj/machinery/cryopod/proc/despawn_occupant()
 	var/mob/living/carbon/human/H = occupant
-	var/list/occupant_organs = H.organs | H.internal_organs
+	var/list/occupant_organs
```
Comment (ACCount12) [inline code/game/machinery/cryopod.dm:289]: Does `W in occupant_organs` work correctly if `occupant_organs` is null?

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -452,6 +452,48 @@ default behaviour is:
 
 	return
 
+// This function is not to be invoked when the mob is killed. It runs when the mob is leaving the game or similar means.
```
Comment (ACCount12) [inline code/modules/mob/living/living.dm:455]: Reenglish?

File: code/game/machinery/cryopod.dm (hunk 2 above)
Comment (TorinoFermic) [reply code/game/machinery/cryopod.dm:289]: The for() will exit early when occupant_organs is null http://www.byond.com/docs/ref/index.html#/proc/for/list

File: code/modules/mob/living/living.dm (hunk 3 above)
Comment (TorinoFermic) [reply code/modules/mob/living/living.dm]: Fixed

**Pull request merged**